rlog/DRAFT-fiat-shamir.mdx

```diff
@@ -0,0 +1,183 @@
+Specifically,
+$\mathsf{Hash} : \{0,1\}^* \rightarrow \{0,1\}^n$.
+
+A cryptographic hash function satisfies the following properties:
+- **Succinct**: The hash function should be easy to compute; the hash $\mathsf{Hash}({\bf{b}})$ can be efficiently computed for any binary string ${\bf{b}}$.
+- **Preimage Resistance**: It should be impossible to work backwards given the output of a hash function. Let ${\bf{y}}$ be a binary string of length $n$.
+    It should be 'impossible' to find some binary string ${\bf{x}}$ so that ${\bf{y}} = \mathsf{Hash}({\bf{x}})$. 
+- **Collision Resistance**: It should be difficult to find two strings that hash to the same value. It should be 'impossible' to find two binary strings ${\bf{x}_1}$ and ${\bf{x}_2}$ so that $\mathsf{Hash}({\bf{x}_1}) = \mathsf{Hash}({\bf{x}_2}).$
+
+Commonly used hash functions include SHA256, Keccak, and Poseidon.
+
+## The Fiat-Shamir heuristic
+The Fiat-Shamir heuristic is the technique used to convert an interactive protocol to a noninteractive protocol.
+This is done by replacing each of the Verifier's messages with a hashed value.
+Specifically, the Prover generates the Verifier's message by evaluating the hash function $\mathsf{Hash}$ with the concatenation of all public values that appear in the protocol thus far.
+If $m_0, \dots, m_t$ denote the public values in the protocol thus far,
+then the Verifier's message is computed as $m_{t+1} := \mathsf{Hash}(m_0|| \cdots || m_t)$.
+
+Since $\mathsf{Hash}$ can be efficiently computed, and the messages $m_0, \dots, m_t$ are public, then any verifying party can compute $m_{t+1}$.
+Critically, since $\mathsf{Hash}$ is preimage resistant and collision resistant,
+the Prover cannot manipulate her choices of the messages $m_0,\dots, m_t$ to influence the message $m_{t+1}$.
+Hence, verifying parties can trust that $m_{t+1}$ is sufficiently random with respect to the preceding messages.
+
+There are two variants of the Fiat-Shamir heuristic: weak and strong.
+The weak variant uses all of the publicly traded messages in computing the Verifier's messages but does not include the public parameters.
+However, in the strong variant all of the publicly traded messages and public parameters are used to compute the Verifier's messages.
+We will provide a discussion on issues that can arise from using the weak Fiat-Shamir heuristic.
+
+### The Schnorr Protocol
+The Schnorr protocol ([9](https://link.springer.com/chapter/10.1007/0-387-34805-0_22)) is usually the first $\Sigma$-protocol that one studies.
+Additionally, the Schnorr protocol can be used as an efficient signature scheme.
+The Schnorr protocol provides a framework that enables the Prover to convince the Verifier that: for group elements $g$ and $X$, the Prover knows the power to raise $g$ to obtain $X$.
+Specifically, the Prover possesses some integer $x$ so that $X = g^x$.
```

Comment:
I'm being punctilious here... but maybe we can add a comment on the additive and the multiplicative notation in cryptography